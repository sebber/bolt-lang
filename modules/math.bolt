fun add(a: Integer, b: Integer): Integer {
    return a + b
}

fun multiply(x: Integer, y: Integer): Integer {
    return x * y
}

fun square(n: Integer): Integer {
    return n * n
}

export add
export multiply
export square