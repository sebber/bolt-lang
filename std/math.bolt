export fun abs(x: Integer): Integer {
    if (x < 0) {
        return 0 - x
    } else {
        return x
    }
}

export fun min(a: Integer, b: Integer): Integer {
    if (a < b) {
        return a
    } else {
        return b
    }
}

export fun max(a: Integer, b: Integer): Integer {
    if (a > b) {
        return a
    } else {
        return b
    }
}