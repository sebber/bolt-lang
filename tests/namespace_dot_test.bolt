import math from "modules/math"

val a := 5
val b := 3
val sum := math.add(a, b)

print(sum)
print("Namespace dot test complete!")