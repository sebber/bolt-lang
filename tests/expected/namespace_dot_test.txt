8
Namespace dot test complete!